code/script.js

```diff
@@ -1,325 +1,570 @@
+  };
+  console.log(category);
+  console.log(value);
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
+  console.log(currentQuestion);
+  let keep;
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+  // Compare the currentQuestion details with the secret person details in a
+  //  different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "hair") {
+    keep = secret.hair === value;
+  } else if (category === "eyes") {
+    keep = secret.eyes === value;
+  } else if (category === "type") {
+    keep = secret.type === value;
+  } else if (category === "accessories") {
+    keep = secret.accessories.includes(value);
+  } else if (category === "weapon") {
+    keep = secret.weapon.includes(value);
   }
-}
 
+  filterCharacters(keep);
+};
 // It'll filter the characters array and redraw the game board.
+
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
+  charactersNotInPlay = charactersInPlay;
+  console.log(charactersNotInPlay);
+  console.log(currentQuestion);
+
+  const keepCharStr = () => {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] === value
+    );
+    charactersNotInPlay = charactersNotInPlay.filter(
+      (person) => person[category] !== value
+    );
+    console.log(charactersInPlay);
+    console.log(charactersNotInPlay);
+    generateBoard();
+  };
+
+  const removeCharStr = () => {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] !== value
+    );
+    charactersNotInPlay = charactersNotInPlay.filter(
+      (person) => person[category] === value
+    );
+    console.log(charactersInPlay);
+    console.log(charactersNotInPlay);
+    generateBoard();
+  };
+
+  const removeCharArray = () => {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => !person[category].includes(value)
+    );
+    charactersNotInPlay = charactersNotInPlay.filter((person) =>
+      person[category].includes(value)
+    );
+    console.log(charactersInPlay);
+    console.log(charactersNotInPlay);
+    generateBoard();
+  };
+
+  const keepCharArray = () => {
+    charactersInPlay = charactersInPlay.filter((person) =>
+      person[category].includes(value)
+    );
+    charactersNotInPlay = charactersNotInPlay.filter(
+      (person) => !person[category].includes(value)
+    );
+    console.log(charactersInPlay);
+    console.log(charactersNotInPlay);
+    generateBoard();
+  };
+
```

Comment:
Love how you are so good at using boolean values and the code here is very easy to read and understand!